src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlDataContract.cs

```diff
@@ -412,6 +435,29 @@ private bool RequiresMemberAccessForCreate(SecurityException securityException)
         }
 #endif
 
+        internal IXmlSerializable ReflectionCreateXmlSerializable(Type type)
+        {
+            if (type.GetTypeInfo().IsValueType)
+            {
+                throw new NotImplementedException("ReflectionCreateXmlSerializable - value type");
+            }
+            else
+            {
+                object o = null;
+                if (type == typeof(System.Xml.Linq.XElement))
+                {
+                    o = new System.Xml.Linq.XElement("default");
+                }
+                else
+                {
+                    ConstructorInfo ctor = GetConstructor();
```

Comment:
XElement is the only type we treat specially when using CodeGen method, see GenerateCreateXmlSerializableDelegate. IXmlSerializable types must have default constructor. The GetConstructor() method would throw if default ctor is not available.